mongomock/filtering.py

```diff
@@ -408,12 +413,27 @@ def func(doc_val, search_val):
     return func
 
 
-def _type_op(doc_val, search_val):
+def _type_op(doc_val, search_val, in_array=False):
```

Comment:
Please do not add the `in_array` arg for now in here. Add it in the PR you need it.

Reply:
Oops... My bad.

Reply:
Hm... Actually, it seems it already is in the correct PR, isn't it?

Reply:
Ah ok, got it. Please make sure to call it with `in_array=True` so that it's explicit below.

Reply:
Done. Thanks!